microsetta_admin/server.py

```diff
@@ -154,39 +154,67 @@ def _search(resource=None):
             return result
 
 
-@app.route('/create_project', methods=['GET', 'POST'])
-def new_project():
-    if request.method == 'GET':
-        return render_template('create_project.html',
-                               **build_login_variables())
-    elif request.method == 'POST':
-        project_name = request.form['project_name']
-        is_microsetta = request.form.get('is_microsetta', 'No') == 'Yes'
-        bank_samples = request.form.get('bank_samples', 'No') == 'Yes'
-        plating_start_date = request.form.get('plating_start_date')
-        if plating_start_date == '':
-            plating_start_date = None
-
-        status, result = APIRequest.post('/api/admin/create/project',
-                                         json={'project_name': project_name,
-                                               'is_microsetta': is_microsetta,
-                                               'bank_samples': bank_samples,
-                                               'plating_start_date':
-                                                   plating_start_date
-                                               })
-
-        if status == 201:
-            return render_template('create_project.html', message='Created!',
-                                   **build_login_variables())
+def _translate_nones(a_dict, do_none_to_str):
+    # Note: this ISN'T a deep copy. This function is NOT set up
+    # for recursing through a multi-layer dictionary
+    result = a_dict.copy()
+    for k, v in result.items():
+        if do_none_to_str and v is None:
+            result[k] = ""
+        elif not do_none_to_str and v == '':
+            result[k] = None
+    return result
+
+
+@app.route('/manage_projects', methods=['GET', 'POST'])
+def manage_projects():
+    result = None
+    if request.method == 'POST':
+        projects_uri = '/api/admin/projects'
+
+        model = {x: request.form[x] for x in request.form}
+        project_id = model.pop('project_id')
+        model['is_microsetta'] = model.get('is_microsetta', False) == 'true'
+        model['bank_samples'] = model.get('bank_samples', False) == 'true'
+        model = _translate_nones(model, False)
+
+        if project_id.isdigit():
```

Comment:
What is the project_id when not project_id.isdigit()?  If None, wouldn't this crash?

Reply:
This is another situation where the available values are determined strictly by the html form.  The `project_id` field in the form (which is where `model` comes from) is a text input that is initialized with a value of an empty string:
`<input type="hidden" name="project_id" id="project_id" value="" />`

IF we are updating an existing project, then the `readRow` javascript function will update the value of this form field to (a string version of) the project id; otherwise, if we are creating a new project, the value will be left as an empty string.  There is not (as far as I know, and should not be :) ) any way for this field to take on any value that is not a string (since the project id is popped off the model BEFORE the model is run through `_translate_nones`, which is used later to turn empty strings into `None`s).